cmd/cue/cmd/testdata/script/cmd_execenv.txtar

```diff
@@ -0,0 +1,34 @@
+exec cue cmd execEnv
+cmp stdout stdout.golden
+
+-- cue.mod/module.cue --
+module: "example.com"
+
+-- stdout.golden --
+usesDefaultsStruct: hello from John Doe
+
+-- foo_tool.cue --
+package foo
+
+import (
+	"tool/cli"
+	"tool/exec"
+)
+
+#message: *"hello" | string
+
+command: execEnv: {
+    usesDefaultsStruct: exec.Run & {
+        cmd: ["sh", "-c", "echo $MESSAGE from $SENDER"]
+        stdout: *"" | string
+        env: {
+            MESSAGE: #message
+            SENDER: *"John Doe" | string
+        }
+    }
```

Comment:
Let me know if there's any concerns with the testing here 🙂 . I noticed an issue with how lists are handled (I think) and opened #2763, so I didn't include a test here for the list variation of env currently.